ee/app/livechat-enterprise/server/hooks/applyDepartmentRestrictions.ts

```diff
@@ -0,0 +1,14 @@
+import { Meteor } from 'meteor/meteor';
+
+import { callbacks } from '../../../../../app/callbacks/server';
+import { addQueryRestrictionsToDepartmentsModel } from '../lib/query.helper';
+import { hasRole } from '../../../../../app/authorization/server/functions/hasRole';
+
+callbacks.add('livechat.applyDepartmentRestrictions', (originalQuery = {}) => {
+	const user = Meteor.user();
```

Comment:
please don't rely on `Meteor.user()` inside callbacks, as it depends on Meteor's context to be set which might not happen some times (depending on the stack of the code)..

and looks like you need only user's `_id`, so what I recommend is to receive user's `_id` as the second argument (which is a "constant" among all callbacks)

Reply:
@murtaza98 please, take a look at Diego's comment. Thanks.

Reply:
Thanks @sampaiodiego, I wasn't aware of this edge case. I've fixed it now. We are no-longer using `Meteor.user()` here now

Reply:
great @murtaza98 , thanks..